src/MyEventEmitter.js

```diff
@@ -1,12 +1,83 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    this.events[eventName].push(listener);
+
+    return this;
+  }
+
+  once(eventName, listener) {
+    const onceWrapper = () => {
+      listener();
+      this.off(eventName, onceWrapper);
+    };
+
+    this.on(eventName, onceWrapper);
+
+    return this;
+  }
+
+  off(eventName, listener) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName].filter(
+      eventListener => eventListener !== listener
+    );
+
+    return this;
+  }
+
+  emit(eventName, ...args) {
+    if (!this.events[eventName]) {
+      return false;
+    }
+
+    this.events[eventName].forEach(listener => listener(...args));
+
+    return true;
+  }
+
+  prependListener(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    this.events[eventName].unshift(listener);
+
+    return this;
+  }
+
+  prependOnceListener(eventName, listener) {
+    const onceWrapper = () => {
+      listener();
+      this.off(eventName, onceWrapper);
+    };
+
+    this.prependListener(eventName, onceWrapper);
+
+    return this;
+  }
+
+  removeAllListeners(eventName) {
+    delete this.events[eventName];
+
+    return this;
+  }
+
+  listenerCount(eventName) {
+    return this.events[eventName].length || 0;
```

Comment:
This will cause TypeError if `this.events[eventName]` is undefined. Simple fix:
```suggestion
    return this.events[eventName]?.length || 0;
```